### DISCUSSION TITLE
How to bind position of `ItemsControl` items on `Canvas` using C# Markup?

### DISCUSSION THREAD

**Discussion by kucint** [unoplatform/uno#16092] [Q&A]
How do I a deal in C# Markup with known issue that elements on `Canvas` does not reflect position correctly until a custom canvas binding is provided?

In XAML, one must provide a code: similar to:
```
      <ItemsControl.ItemContainerStyle>
        <Style TargetType="ContentPresenter">
          <Setter Property="internal:CanvasBindingHelper.CanvasLeftBindingPath" Value="X"/>
          <Setter Property="internal:CanvasBindingHelper.CanvasTopBindingPath" Value="Y"/>
        </Style>
      </ItemsControl.ItemContainerStyle>
```

where `CanvasBindingHelper` is a custom converter class that binds properties of ItemsControl items (typically X and Y) with Canvas properties `Left` and `Top`.

what is equivalent in C# Markup?

**Comment by francoistanguay** [comment]
I thought this had been answered previously by the link to this PR if the question is really about how to set Attached Properties in a Style? If so, let's update the Title as well.
 
https://github.com/unoplatform/uno.extensions/pull/2218/files

**Reply by dansiegel** [reply]
@kucint does the updated docs that @francoistanguay linked to help? Feel free to ping me in Discord as well if you'd like to discuss in more detail.

**Reply by kucint** [reply]
Hi @dansiegel, I can't actually test it till fix for #16091 is not released.
I will test it as soon as possible and let you know.
